### DISCUSSION TITLE
Missing `application` property in `OgmaServiceMeta` type

### DISCUSSION THREAD

**Discussion by tukusejssirs** [jmcdo29/ogma#1377] [Q&A] [answered]
According to [`packages/logger/README.md`](https://github.com/jmcdo29/ogma/blob/main/packages/logger/README.md#adding-context-and-application-name), one can change `application` value the second parameter of any log method.

However, in Nest, it won’t suggest to add this property. When I provide it, Nest nor TypeScript won’t throw an error though.

I think the issue lays in [`packages/nestjs-module/src/interfaces/ogma-service-meta.interface.ts`](https://github.com/jmcdo29/ogma/blob/main/packages/nestjs-module/src/interfaces/ogma-service-meta.interface.ts):

```ts
export interface OgmaServiceMeta {
  context?: string;
  correlationId?: string;
  [key: string]: unknown;
}
```

I understand that `[key: string]: unknown` accepts `application` property, but it might be a good idea to list all properties that are valid for `OgmaServiceMeta` instead.

Anyway, what is `correlationId` for? Somehow I missed any mention of it in the docs.

**Comment by jmcdo29** [comment, marked as answer]
Huh, sure enough. Wonder if I did that cause I never thought that application would be changed within a Nest application. Either way, it can be added.

`correlationId` is a way to correlate the logs to each other, also commonly called a `requestId`. All the logs with the same correlation id mean that those logs were made as a part of the same request/command/instruction/etc. Most of the time used in logregation services like LogRocket or DataDog to give extra metrics about your logs as well.

**Reply by tukusejssirs** [reply]
> Wonder if I did that cause I never thought that application would be changed within a Nest application.

I don’t say it makes sense (most of the time it does not IMHO), however, it is possible. :wink:

And thanks for the explanation how `correclationId` is (can be) used.

> Either way, it can be added.

Do you want to add it yourself? :wink:

Adding a property would not be an issue for me, however, maybe you want to remove `[key: string]: unknown;` altogether, update `@ogma/nestjs-module` docs to mention this. :wink:

**Reply by jmcdo29** [reply]
Feel free to just add `application?: string` to the interface you pointed out. And keep the `[key: string]: unknown`, so that any other metadata people want to pass can be passed without Typescript errors
